applications/SwimmingDEMApplication/python_scripts/swimming_DEM_solver.py

```diff
@@ -0,0 +1,235 @@
+        self.calculating_fluid_in_current_step = bool(time >= self.next_time_to_solve_fluid)
+        if self.calculating_fluid_in_current_step:
+            self.next_time_to_solve_fluid = self.fluid_solver.AdvanceInTime(time)
+            self.fluid_step += 1
+
+        return self.step, self.time
+
+    def UpdateALEMeshMovement(self, time): # TODO: move to derived solver
+        if self.project_parameters["ALE_option"].GetBool():
+            self.rotator.RotateMesh(self.fluid_solver.main_model_part, time)
+            self._GetProjectionModule().UpdateDatabase(self.CalculateMinElementSize())
+
+    def CalculateMinElementSize(self):
+        return self.h_min
+
+    def AssessStationarity(self):
+        Say("Assessing Stationarity...\n")
+        self.stationarity = self.stationarity_tool.Assess(self.fluid_solver.main_model_part)
+        self.stationarity_counter.Deactivate(self.stationarity)
+
+    def ComputePostProcessResults(self):
+        if self.project_parameters["coupling_level_type"].GetInt():
+            self._GetProjectionModule().ComputePostProcessResults(self.dem_solver.spheres_model_part.ProcessInfo)
+
+    def CannotIgnoreFluidNow(self):
+        return self.solve_system and self.calculating_fluid_in_current_step
+
+    def Predict(self):
+        if self.CannotIgnoreFluidNow():
+            self.fluid_solver.Predict()
+
+    def ApplyForwardCoupling(self, alpha='None'):
+        self._GetProjectionModule().ApplyForwardCoupling(alpha)
+
+    def ApplyForwardCouplingOfVelocityToSlipVelocityOnly(self, time=None):
+        self._GetProjectionModule().ApplyForwardCouplingOfVelocityToSlipVelocityOnly()
+
+    def _GetProjectionModule(self):
+        if not hasattr(self, 'projection_module'):
+            self.projection_module = self._ConstructProjectionModule()
+        return self.projection_module
+
+    def SolveSolutionStep(self):
+        # update possible movements of the fluid mesh
+        self.UpdateALEMeshMovement(self.time)
+
+        # Solving the fluid part
+        Say('Solving Fluid... (', self.fluid_solver.main_model_part.NumberOfElements(0), 'elements )\n')
+        self.solve_system = not self.project_parameters["fluid_already_calculated"].GetBool() and not self.stationarity
+
+        if self.CannotIgnoreFluidNow():
+            self.SolveFluid()
+        else:
+            Say("Skipping solving system for the fluid phase...\n")
+
+        # Check for stationarity: this is useful for steady-state problems, so that
+        # the calculation stops after reaching the solution.
+        if self.stationarity_counter.Tick():
+            self.AssessStationarity()
+
+        # Compute nodal quantities to be printed that are not generated as part of the
+        # solution algorithm. For instance, the pressure gradient, which is not used for
+        # the coupling but can be of interest.
+        self.ComputePostProcessResults() # TODO: Make sure this is run only when needed
```

Comment:
We already moved this into the analysis stage

Reply:
Removed.